state-chain/pallets/cf-elections/src/vote_storage/change.rs

```diff
@@ -0,0 +1,85 @@
+use codec::{Decode, Encode};
+use frame_support::{
+	pallet_prelude::{Member, TypeInfo},
+	Parameter,
+};
+
+use super::{AuthorityVote, VoteComponents, VoteStorage};
+
+use crate::{CorruptStorageError, SharedDataHash};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, TypeInfo)]
+pub struct MonotonicChangeVote<Value, BlockHeight> {
+	pub value: Value,
+	pub block: BlockHeight,
+}
+pub struct MonotonicChange<T: Parameter + Member, S: Parameter + Member> {
+	_phantom: core::marker::PhantomData<(T, S)>,
+}
+impl<T: Parameter + Member, S: Parameter + Member> VoteStorage for MonotonicChange<T, S> {
+	type Properties = ();
+	type Vote = MonotonicChangeVote<T, S>;
+	type PartialVote = MonotonicChangeVote<T, S>;
+	type IndividualComponent = S;
+	type BitmapComponent = T;
+	type SharedData = ();
+
+	fn vote_into_partial_vote<H: FnMut(Self::SharedData) -> SharedDataHash>(
+		vote: &Self::Vote,
+		mut _h: H,
+	) -> Self::PartialVote {
+		(*vote).clone()
```

Comment:
This means we don't have a partial vote, and we always vote the full vote. The plan is to extend this electoral system for the contract witnessing, and so there the value would be much larger in size - and we don't want every validator submitting all that data - we should probably store a hash of the value in the bitmap, and then construct the full vote from the shared data